### PR TITLE
add a table for more convenient access to docker container environment

### PR DESCRIPTION
**Pull request by nabilschear**
All the docker container environment variables are joined with a comma and embedded into a single field in the `docker_containers` table.  This makes it hard to construct queries against individual variables.  This change introduces a new table called `docker_container_envs` that splits these environment variables out in separate rows so that they can be used in a similar manner to the `process_envs` table.

Structured in a similar manner to the `docker_container_labels` table and re-using the variable parsing code from `process_envs`

### CODE REVIEW

**Comment by mike-myers-tob** [comment]
Hi @nabilschear !

It looks like the integration test is commented out, was this intentional?

**Comment by nabilschear** [comment]
@mike-myers-tob nearly all the docker_* tables have this very simple structure for their tests.  They are basically smoketests to make sure that the query can run (but don't check any returned values).  See https://github.com/osquery/osquery/blob/master/tests/integration/tables/docker_image_labels.cpp as an example.  In my case, I copied from this file: https://github.com/osquery/osquery/blob/master/tests/integration/tables/docker_container_labels.cpp

**Comment by nabilschear** [comment]
any updates on this one?

File: specs/posix/docker_container_envs.table [hunk 1]
```diff
@@ -0,0 +1,13 @@
+table_name("docker_container_envs")
+description("Docker container environment variables.")
+schema([
+    Column("id", TEXT, "Container ID", index=True),
+    Column("key", TEXT, "Environment variable name", index=True),
```
[the remark this answers is not in the document]
Comment (nabilschear) [reply specs/posix/docker_container_envs.table]: I removed indexed from `key`.  It now has a similar structure to the `process_envs` table, which I think is right (replacing `pid` for `id`)

**Review by directionless — APPROVED** [review]
While I'm not totally sure we need it, it seems reasonably in keeping with the rest of the docker tables. Approved, pending CI

**Comment by directionless** [comment]
I suspect the build failures are because of the LE cert expiration. That should be fixed on HEAD. Can you rebase please?

**Comment by Smjert** [comment]
> I suspect the build failures are because of the LE cert expiration. That should be fixed on HEAD. Can you rebase please?

Closing and reopening the PR (as Mike previously did) automatically updates the temporary merge branch that Github uses with the CI behind the curtains, aligning it to master.

**Pull request merged**